compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -2506,6 +2506,20 @@ setWarpReductionConfig(IREE::GPU::TargetAttr target,
 
   SmallVector<int64_t> workgroupTileSizes(op.getNumParallelLoops(), 1);
 
+  int64_t reductionSize = 1;
+  for (int64_t dim : reductionDims)
+    reductionSize *= bounds[dim];
```

Comment:
is this safe? previously this was guarded by the dynamic dim count logic below

```
      for (unsigned dim : reductionDims) {
        if (ShapedType::isDynamic(bounds[dim])) {
```

may be needed here too so you don't end up with a negative size by multiplying by -1?

Reply:
https://github.com/iree-org/iree/pull/21063/files#diff-e491707653524e61bd22ba47ca5bd8414065b70bd1c6b85f45947929d7a49aafL2476 already checks that none of the reduction dimensions are dynamic. 
This all is going to be deleted soon anyway.